odoo/models.py

```diff
@@ -5183,11 +5270,44 @@ def _parent_store_update(self):
             self.env.cache.update(records, self._fields['parent_path'], updated.values())
             records.modified(['parent_path'])
 
+    def _clean_properties(self) -> None:
+        """ Remove all properties of ``self`` that are no longer in the related definition """
+        for fname, field in self._fields.items():
+            if field.type != 'properties':
+                continue
+            for record in self:
+                old_value = record[fname]
+                if not old_value:
+                    continue
+
+                definitions = field._get_properties_definition(record)
+                all_names = {definition['name'] for definition in definitions}
+                new_values = {name: value for name, value in old_value.items() if name in all_names}
+                if len(new_values) != len(old_value):
+                    record[fname] = new_values
+
     def _load_records_write(self, values):
```

Comment:
since this method `ensure_one`, maybe we should rename it `_load_record_write`?

Reply:
I am agree with you with the principle but AFAIK this feature is for the V18 and I don't want to create a upgrade PR for that (https://github.com/odoo/upgrade/blob/a8a0740e3d17c9bd5c86f5e603cb2a02c693fad6/migrations/base/0.0.0/pre-models-ir_ui_view.py#L997). We will refactor the export/import in the next version normally.